docs/apidiffs/current_vs_latest/opentelemetry-semconv.txt

```diff
@@ -0,0 +1,170 @@
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> NETWORK_TRANSPORT
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> NETWORK_PROTOCOL_NAME
++++  NEW CLASS: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.semconv.NetworkAttributes$NetworkTransportValues  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String TCP
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String UDP
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String QUIC
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String PIPE
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String UNIX
++++  NEW CLASS: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.semconv.NetworkAttributes$NetworkTypeValues  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String IPV6
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String IPV4
++++  NEW CLASS: PUBLIC(+) FINAL(+) io.opentelemetry.semconv.OtelAttributes  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> OTEL_SCOPE_NAME
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> OTEL_STATUS_DESCRIPTION
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> OTEL_STATUS_CODE
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> OTEL_SCOPE_VERSION
++++  NEW CLASS: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.semconv.OtelAttributes$OtelStatusCodeValues  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String ERROR
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String OK
++++  NEW CLASS: PUBLIC(+) FINAL(+) io.opentelemetry.semconv.SchemaUrls  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_24_0
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_23_1
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_22_0
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_29_0
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_28_0
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_27_0
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_26_0
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String V1_25_0
```

Comment:
any thoughts on this list (i.e. how far to go back)?

cc @lmolkova

Reply:
Notable versions:

- 1.21.0 was first release from `semantic-conventions` repo
- 1.23.0 stable http conventions
- 1.24.0 stable JVM conventions, significant restructuring of `semantic-convention-java` to switch to `{Domain}Attributes` naming convention and breakout `opentelemetry-semconv` and `opentelemetry-semconv-incubating` artifacts. 
- 1.28.0 switch code gen for `semantic-convention-java` to weaver

We could start conservative with the schema URLs we include and go further back later if needed?

Reply:
yeah, I don't really have any specific concerns or objection with the existing list

> 1.21.0 was first release from `semantic-conventions` repo

this seems as good as any reason, but agree we can add later, only thing that would affect this PR / release would be if we wanted to restrict the list

> 1.23.0 stable http conventions

btw, I noticed `1.23.0` is missing in the list (but `1.22.0` and `1.23.1` are included)

Reply:
I did a quick search for [`"SchemaUrls.V1_" NOT is:fork`](https://github.com/search?q=%22SchemaUrls.V1_%22+NOT+is%3Afork&type=code) across GitHub and it seems there are just a few places that use it.

I only see usages of V1_24_0 and above - which makes sense - we ask to stay at 1.24.0 in DB and messaging migration plan.

I'd remove anything below 1.24.0.

Reply:
ok, I'll send a PR to remove 1.22.0 and 1.23.1

Reply:
#134

Reply:
Quick thought - would we regret having exact patch in the version?
Should we do `V1_25` (`V1_25_X`) instead similarly to release branches?

I.e. if semconv is patched, we'd not add a new schema version.  
Happy to send a PR if you folks think it makes sense.

Reply:
I kind of agree, but if we're saying there's no difference between patched versions, shouldn't the schema url itself be `https://opentelemetry.io/schemas/1.29` or `https://opentelemetry.io/schemas/1.29.x`?

Reply:
That's a can of worms though because these are resolvable URLs, and `../1.29` and `../1.29.x` do not resolve. I think our schema URLs should reflect whatever the publishing convention is on opentelemetry.io. We could open an issue about this and block until resolved, but I don't think we should block stability. If opentelemetry.io later updates to reflect one of the conventions you propose, we can update our schema URL values from that point onward.

Reply:
ok, I created https://github.com/open-telemetry/opentelemetry-specification/issues/4355 - we can change the format of the constant later if it comes through.